plugin/kubernetes/external.go

```diff
@@ -19,6 +19,10 @@ func (k *Kubernetes) External(state request.Request) ([]msg.Service, int) {
 	segs := dns.SplitDomainName(base)
 	last := len(segs) - 1
 	if last < 0 {
+		// this is a request for the zone itself. Return all services that are in exposed namespaces
```

Comment:
this is only correct if there is an ixfr/axfr request, this should return the apex of the zone, not all records.

Reply:
I guess that is the responsibility of the caller to reply the correct records depending IXFR/AXFR.
this function `External` here is only to return Services.